### PR TITLE
Add Checkpoints with history

### PR DESCRIPTION
**Pull request by rofinn**
Package: https://github.com/invenia/Checkpoints.jl

This is part of the Contribution2020 event, where Invenia (and others) are open sourcing a bunch of stuff. 
In this particular case, we are taking the whole package from closed to open, and including the history of 7 past releases. 
This PR involved copying the `C/Checkpoints` files from our private registry and adding our `Registry.toml` line.
No other manual changes have been made. Since we're manually copying the registry entries over this will likely fail the `AutoMerge` check and require manual intervention from the maintainers.

Similar to https://github.com/JuliaRegistries/General/pull/25988

### CODE REVIEW

File: C/Checkpoints/Deps.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+["0-0.1"]
+AWSCore = "4f1ea46c-232b-54a6-9b17-cc2d0f3e6598"
+AWSSDK = "0d499d91-6ae5-5d63-9313-12987b87d5ad"
+BSON = "fbb218c0-5317-5bc6-957e-2ee96dd4b1f0"
+Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
+Mocking = "78c3b35d-d492-501b-9361-3d52fe80e533"
+
+["0-0.2.1"]
+AWSTools = "83bcdc74-1232-581c-948a-f29122bf8259"
```
Comment (rofinn) [inline C/Checkpoints/Deps.toml:14]: Looks like the history will require that AWSTools gets registered in General first.

Comment (rofinn) [reply C/Checkpoints/Deps.toml:14]: Should work after https://github.com/JuliaRegistries/General/pull/26003 is merged.

**Comment by DilumAluthge** [comment]
Don't worry about the AutoMerge checks.

But we do need the CI checks to pass

**Comment by rofinn** [comment]
> But we do need the CI checks to pass.

Yeah, this needs https://github.com/JuliaRegistries/General/pull/26003 to be merged for CI checks to pass.

**Comment by rofinn** [comment]
Thanks @DilumAluthge!

**Pull request merged**